src/sentry/static/sentry/app/views/stream/stream.jsx

```diff
@@ -96,6 +96,20 @@ const Stream = createReactClass({
   },
 
   componentWillMount() {
+    const organization = this.getOrganization();
+    const hasSentry10 = new Set(organization.features).has('sentry10');
+
+    if (hasSentry10) {
+      const project = this.getProject();
+      const {location} = this.props;
+      const query = qs.parse(location.search);
```

Comment:
Do you need to parse if you just read from `location.query`?

Reply:
I was worried about mutating `location.query` in-place, and wasn't confident that it would be side-effect free.